sdk/react/ooyalaSkinCore.js

```diff
@@ -133,7 +133,24 @@ OoyalaSkinCore.prototype.shouldShowLandscape = function() {
 };
 
 OoyalaSkinCore.prototype.shouldShowDiscoveryEndscreen = function() {
-  return (this.skin.state.upNextDismissed == true && this.skin.props.endScreen.screenToShowOnEnd == "discovery");
+  // Only care if discovery on endScreen should be shown
+  if (this.skin.props.endScreen.screenToShowOnEnd !== "discovery") {
```

Comment:
pushed a change for what you're saying. This is why we need testing in place. We'll have to work towards it little by little. Thanks!